UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/Interfaces/INMDataService.cs

```diff
@@ -1,12 +1,13 @@
-﻿using UKHO.MaritimeSafetyInformation.Common.Models.NoticesToMariners;
+﻿using Microsoft.AspNetCore.Mvc.Rendering;
+using UKHO.MaritimeSafetyInformation.Common.Models.NoticesToMariners;
 
 namespace UKHO.MaritimeSafetyInformation.Web.Services.Interfaces
 {
     public interface INMDataService
     {
         Task<List<ShowFilesResponseModel>> GetWeeklyBatchFiles(int year, int week, string correlationId);
         Task<List<ShowDailyFilesResponseModel>> GetDailyBatchDetailsFiles(string correlationId);
-        List<KeyValuePair<string, string>> GetAllYears(string correlationId);
-        List<KeyValuePair<string, string>> GetAllWeeksOfYear(int year, string correlationId);
+        Task<List<YearWeekModel>> GetAllYearWeek(string correlationId);
```

Comment:
This is currently not used outside of the class `NMDataService`:
```
        Task<List<YearWeekModel>> GetAllYearWeek(string correlationId);
```

Reply:
We are using this in GetWeeklyFilesResponseModelsAsync

Reply:
Yes, which is internal to `NMDataService`, so not needed in the interface.

Reply:
removed GetAllYearWeek from interface